data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/StartsWithExpressionFunctionTest.java

```diff
@@ -0,0 +1,87 @@
+package org.opensearch.dataprepper.expression;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.opensearch.dataprepper.expression.StartsWithExpressionFunction.STARTS_WITH_FUNCTION_NAME;
+
+public class StartsWithExpressionFunctionTest {
+
+    private Event testEvent;
+
+    @BeforeEach
+    void setup() {
+
+    }
+
+    private Event createTestEvent(final Object data) {
+        return JacksonEvent.builder().withEventType("event").withData(data).build();
+    }
+
+    private ExpressionFunction createObjectUnderTest() {
+        return new StartsWithExpressionFunction();
+    }
+
+    @ParameterizedTest
+    @MethodSource("ValidStartsWithProvider")
+    void startsWith_returns_expected_result_when_evaluated(
+            final String value, final String prefix, final boolean expectedResult) {
+        final String key = "test_key";
+        testEvent = createTestEvent(Map.of(key, value));
+
+        final ExpressionFunction objectUnderTest = createObjectUnderTest();
+        assertThat(objectUnderTest.getFunctionName(), equalTo(STARTS_WITH_FUNCTION_NAME));
+
+        final Object result = objectUnderTest.evaluate(List.of("/" + key, "\"" + prefix + "\""), testEvent, mock(Function.class));
```

Comment:
It looks like you are supporting a key for the substring part. Add a unit test for that as well.